src/main/java/com/nvidia/spark/rapids/jni/CaseWhen.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids.jni;
+
+import ai.rapids.cudf.*;
+
+
+/**
+ * Exedute SQL `case when` semantic.
+ * If there are multiple branches and each branch uses scalar to generator value,
+ * then it's fast to use this class because it does not generate temp string columns.
+ *
+ * E.g.:
+ *   SQL is:
+ *     select
+ *        case
+ *          when bool_1_expr then "value_1"
+ *          when bool_2_expr then "value_2"
+ *          when bool_3_expr then "value_3"
+ *          else "value_else"
+ *        end
+ *      from tab
+ *
+ * Execution steps:
+ *   Execute bool exprs to get bool columns, e.g., gets:
+ *     bool column 1: [true,  false, false, false]  // bool_1_expr result
+ *     bool column 2: [false, true,  false, flase]  // bool_2_expr result
+ *     bool column 3: [false, false, true,  flase]  // bool_3_expr result
+ *   Execute `selectFirstTrueIndex` to get the column index for the first true in bool columns.
```

Comment:
Here we have 4 options instead of 3 (the `value_else` should be considered). As such, we can generate the 4th bool column containing all-true values. Doing so, we won't need to launch another kernel to set `value_else` after gathering values from the first 3 columns.